src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceService.scala

```diff
@@ -412,6 +412,31 @@ class ResourceService(
     eventualSubjects.toList.sequence.map(_.toMap)
   }
 
+  //Disallow orphaning, leaving when it's via a group, or leaving when the resource is public
+  def leaveResource(resourceType: ResourceType, resourceId: FullyQualifiedResourceId, samUser: SamUser, samRequestContext: SamRequestContext): IO[List[Boolean]] = {
+    accessPolicyDAO.listAccessPolicies(resourceId, samRequestContext) flatMap { policiesForResource =>
+      //1. Make sure that there are no public policies for this resource. Leaving a public resource is not supported.
```

Comment:
Is this restriction due to some API limitation? If a public workspace has users with direct write/admin access, should they not be able to give up those permissions?

Reply:
That's a fair point- this could be handled similarly to how we handle the case of removing direct membership when there is also indirect membership.